sql/analyzer/resolve_column_defaults.go

```diff
@@ -173,19 +173,33 @@ func resolveColumnDefaults(ctx *sql.Context, _ *Analyzer, n sql.Node, _ *plan.Sc
 			col := sch[index]
 
 			eWrapper := expression.WrapExpression(node.Default)
-			newExpr, same, err := resolveColumnDefault(ctx, col, eWrapper)
+			newExpr, sameColumnDefault, err := resolveColumnDefault(ctx, col, eWrapper)
 			if err != nil {
 				return node, transform.SameTree, err
 			}
-			if same {
-				return node, transform.SameTree, nil
-			}
 
 			newNode, err := node.WithDefault(newExpr)
 			if err != nil {
 				return node, transform.SameTree, err
 			}
-			return newNode, transform.NewTree, nil
+
+			// After resolving the new column default value, ensure all column defaults in the target schema are resolved
+			newNode, sameTargetSchema, err := transform.OneNodeExpressions(newNode, func(e sql.Expression) (sql.Expression, transform.TreeIdentity, error) {
```

Comment:
Seems like this call to `transform.OneNodeExpressions` should also handle the logic above it. AlterDefaultSet should implement Expressioner to include its column default as well as the defaults of the target schema so that this works. If that doesn't work for some reason, add a note about why

Reply:
Oooh... good idea. That block felt pretty messy. I'll give this a shot!